comp/logs/agent/config/go.mod

```diff
@@ -0,0 +1,81 @@
+module github.com/DataDog/datadog-agent/comp/logs/agent/config
+
+go 1.21
+
+replace (
+	github.com/DataDog/datadog-agent/cmd/agent/common/path => ../../../../cmd/agent/common/path
```

Comment:
Does it have to be a 1 to 1 relation between a replace and a require? I ask this, because I do not see the `require github.com/DataDog/datadog-agent/cmd/agent/common/path` in this file

Reply:
Also missing `pkg/version`

Reply:
It doesn't have to be 1 to 1, the replace is needed because this module is imported by the `comp/core/config` dependency. This is the output if removed:
```
❯ go mod tidy
go: downloading github.com/DataDog/datadog-agent/cmd/agent/common/path v0.0.0-00010101000000-000000000000
go: github.com/DataDog/datadog-agent/comp/logs/agent/config tested by
        github.com/DataDog/datadog-agent/comp/logs/agent/config.test imports
        github.com/DataDog/datadog-agent/comp/core/config tested by
        github.com/DataDog/datadog-agent/comp/core/config.test imports
        github.com/DataDog/datadog-agent/cmd/agent/common/path: github.com/DataDog/datadog-agent/cmd/agent/common/path@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

Reply:
Why doesn't it appear in the `indirect` then ?

Reply:
Modules `replace`'d which aren't `require`'d:
- cmd/agent/common/path
- comp/core/flare/types
- pkg/telemetry

Modules `require`'d which are not `replace`'d:
- pkg/util/filesystem
- pkg/util/scrubber
- pkg/util/system/socket
- pkg/version

Reply:
We discussed on slack and found that the modules that need to be replaced but aren't required are more recent than the last release, as indicated in this [comment](https://github.com/golang/go/issues/28397#issuecomment-433715034). This behavior was consistent when @pgimalac tested locally.

I added replace statements for the modules that were required but not replaced.